aiida/engine/processes/workchains/restart.py

```diff
@@ -202,10 +236,11 @@ def inspect_process(self):  # pylint: disable=too-many-branches
         last_report = None
 
         # Sort the handlers with a priority defined, based on their priority in reverse order
-        for handler in sorted(self.get_process_handlers(), key=lambda handler: handler.priority, reverse=True):
+        get_priority = lambda handler: handler.priority
+        for handler in sorted(self.get_process_handlers(), key=get_priority, reverse=True):
```

Comment:
why this change?

Reply:
because `handler` is a wrapped function with dynamically added attributes (in this case `priority`). There is no way to properly type these (same with decorated function processes) and so mypy was failing

Reply:
Could have used a comments perhaps, because I am sure I will at some point try to make it a one-liner again and be faced with a failing pre-commit. Fine to leave as is though

Reply:
fair, can't be bothered now though lol
technically I find these design patterns, where you are dynamically adding a new method a bit problematic, same with how the function processes are implemented. But I see the thinking behind it: being able to use `functools.wrap`